fjord/suggest/providers/trigger/provider.py

```diff
@@ -0,0 +1,112 @@
+import logging
+
+from fjord.base.google_utils import ga_track_event
+from fjord.redirector import (
+    Redirector,
+    RedirectParseError,
+    build_redirect_url
+)
+from fjord.suggest import Link, Suggester
+from fjord.suggest.providers.trigger.models import TriggerRule
+
+
+PROVIDER = 'trigger'
+PROVIDER_VERSION = 1
+
+TRIGGER = 'tgr'
+EVENT_CATEGORY = 'trigger' + str(PROVIDER_VERSION)
+SESSION_KEY = 'trigger_{0}'
+
+
+logger = logging.getLogger('i.trigger')
+
+
+def format_redirect(slug):
+    return '{0}.{1}'.format(TRIGGER, slug)
+
+
+def parse_redirect(redirect):
+    try:
+        trigger, slug = redirect.split('.')
+        return trigger, slug
```

Comment:
This is because the code I copied it from converted the second part into an int and had exception handling for that. I didn't think of the fact that this enforces it to be two parts--that's a good reason to leave it as is.